Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesStepTransitionAnimationTrait.swift

```diff
@@ -17,9 +21,10 @@ internal class AppcuesStepTransitionAnimationTrait: ContainerDecoratingTrait {
     private let duration: TimeInterval
     private let easing: Easing
 
-    required init?(config: DecodingExperienceConfig, level: ExperienceTraitLevel) {
-        self.duration = config["duration"] ?? 0.3
-        self.easing = config["easing"] ?? .linear
+    required init?(configuration: ExperiencePluginConfiguration, level: ExperienceTraitLevel) {
+        let config = configuration.decode(Config.self)
```

Comment:
I like the consistency of `guard let config = configuration.decode(Config.self)`  in every trait & action init, even if the decode will never fail (like in this case).

Reply:
I _think_ we can do that in most or all. I just don't want to accidentally fail something in the case of: config in it's entirety is optional (can go with all defaults), but if it is not-omitted, then it has some non-optional values. Need to make sure all the cases are covered to let it gracefully fall back to defaults in such cases. Need to review all the usages for that.

Reply:
After some discussion, leaving as-is for now.  This way the config can capture the required/optional json properties. It is up to the trait/action implementor to declare if completely missing config is ok or not, for that particular case.